File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    bool isMul(string s){
        return s=="+" || s=="-" || s=="*" || s=="/";
    }
    int op(int a, int b, string s){
        if(s=="+") return a+b;
        if(s=="-") return a-b; 
        if(s=="*") return a*b; 
        else return a/b; 
    }
    int evalRPN(vector<string>& tokens) {
        int a,b;
        stack<int> stk;
        for(auto s: tokens){
            if(isMul(s)){
                b=stk.top();
                stk.pop();
                a=stk.top();
                stk.pop();
                stk.push(op(a,b,s));
            }
            else{
                stk.push(stoi(s));
            }
        }
        return stk.top();
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    private int calculate(int a, int b, String s){
        if(s.equals("+")) return a+b;
        else if(s.equals("-")) return a-b;
        else if(s.equals("*")) return a*b;
        else return a/b;
    }
    public int evalRPN(String[] tokens) {
        Set<String> sign = new HashSet<>(Arrays.asList("+","-","*","/"));
        Stack<Integer> stk = new Stack<>();
        for(int i=0;i<tokens.length;i++){
            String tkn= tokens[i];
            if(!sign.contains(tkn)){
                stk.push(Integer.valueOf(tkn));
            }
            else{
                int b=stk.pop(), a=stk.pop();
                stk.push(calculate(a,b,tkn));
            }
        }
        return stk.peek();
    }
}